docs/cpp_docs.rst

```diff
@@ -26,94 +44,556 @@ The beginning of both header and source code files should include
 
       // -*- lsst-c++ -*-
 
-2. A copyright and license block
+2. A copyright and license block (note: **NOT** a Doxygen comment block)
 
    .. code-block:: cpp
 
       /*
-      * LSST Data Management System
-      * See COPYRIGHT file at the top of the source tree.
-      *
-      * This product includes software developed by the
-      * LSST Project (http://www.lsst.org/).
-      *
-      * This program is free software: you can redistribute it and/or modify
-      * it under the terms of the GNU General Public License as published by
-      * the Free Software Foundation, either version 3 of the License, or
-      * (at your option) any later version.
-      *
-      * This program is distributed in the hope that it will be useful,
-      * but WITHOUT ANY WARRANTY; without even the implied warranty of
-      * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-      * GNU General Public License for more details.
-      *
-      * You should have received a copy of the LSST License Statement and
-      * the GNU General Public License along with this program. If not,
-      * see <http://www.lsstcorp.org/LegalNotices/>.
-      */
+       * LSST Data Management System
+       * See COPYRIGHT file at the top of the source tree.
+       *
+       * This product includes software developed by the
+       * LSST Project (http://www.lsst.org/).
+       *
+       * This program is free software: you can redistribute it and/or modify
+       * it under the terms of the GNU General Public License as published by
+       * the Free Software Foundation, either version 3 of the License, or
+       * (at your option) any later version.
+       *
+       * This program is distributed in the hope that it will be useful,
+       * but WITHOUT ANY WARRANTY; without even the implied warranty of
+       * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+       * GNU General Public License for more details.
+       *
+       * You should have received a copy of the LSST License Statement and
+       * the GNU General Public License along with this program. If not,
+       * see <http://www.lsstcorp.org/LegalNotices/>.
+       */
 
-File Description Comment for Header Files
-=========================================
+.. _cpp-doxygen-basics:
+
+Basic Format of Documentation Blocks
+=====================================
+
+This page covers the most important concepts in writing Doxygen-friendly C++ comments for the LSST Stack, though `Doxygen's manual <http://www.doxygen.org/manual/>`_ is the most comprehensive reference.
+
+.. _cpp-doxygen-javadoc:
+
+Documentation MUST be delimited in Javadoc style
+-------------------------------------------------
+
+Documentation blocks **must** begin with ``/**`` and end in ``*/``. The only exception is for :ref:`inline parameter documentation <cpp-doxygen-parameters-inline>`, which may begin with ``///<``.
```

Comment:
I have no ab initio preference between `///` and `/** ... */`, but I think the former is more common in the codebase.